### PR TITLE
Sumator

### PR DESCRIPTION
**Pull request by Stanislava88**
first task invalid input exception

### CODE REVIEW

File: src/sumator/sumString.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package sumator;
+
+import com.sun.org.apache.xpath.internal.SourceTree;
+
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class sumString {
```
Comment (mgenov) [inline src/sumator/sumString.java]: Class name should start with upper case character.

File: src/sumator/sumString.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  public void sum(String a, String b) {
+    try {
+      int x = Integer.parseInt(a);
+      int y = Integer.parseInt(b);
+      int sum = x + y;
+      System.out.println("The sum of number is:" + sum);
```
Comment (mgenov) [inline src/sumator/sumString.java]: If you follow the rule for "Print" statements, you will be forced to change signature of sum method to return number value.

**Comment by Stanislava88** [comment]
fixed

File: src/sumator/Demo.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class Demo {
+  public static void main(String[] args) {
+    try {
+      SumString sum = new SumString();
+      String a = new String("15");
+      String b = new String("20");
+      sum.sum(a, b);
+    } catch (NullPointerException e) {
```
Comment (mgenov) [inline src/sumator/Demo.java]: Are you sure that this will happen ?

File: src/sumator/SumString.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ */
+public class SumString {
+  public int sum(String a, String b) {
+    try {
+      int x = Integer.parseInt(a);
+      int y = Integer.parseInt(b);
+      int sum = x + y;
+      return sum;
+    } catch (NumberFormatException e) {
+      return 101;
```
Comment (mgenov) [inline src/sumator/SumString.java]: You can throw IllegalArgumentException here, to tell the caller that wrong arguments are passed.

File: src/sumator/SumString.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package sumator;
+
+import com.sun.org.apache.xpath.internal.SourceTree;
+
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class SumString {
```
Comment (mgenov) [inline src/sumator/SumString.java]: SumString is "verb", not noun and need to be improved.

File: src/sumator/Sumator.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+public class Sumator {
+  public int sum(String a, String b) {
+    try {
+      int x = Integer.parseInt(a);
+      int y = Integer.parseInt(b);
+      int sum = x + y;
+      return sum;
+    } catch (NumberFormatException e) {
+      throw new IllegalArgumentException("Invalid input");
+    } catch (Exception e) {
```
Comment (mgenov) [inline src/sumator/Sumator.java]: By using it that way, you are omitting the stack trace. 

http://docs.oracle.com/javase/7/docs/api/java/lang/IllegalArgumentException.html#IllegalArgumentException(java.lang.String,%20java.lang.Throwable)

**Comment by Stanislava88** [comment]
i think i fixed it

**Comment by mgenov** [comment]
The idea was to take a look what is StackTrace and how it's used, cause the topic is Exceptions. Your code now looks better, cause you are using RegEx for validation which is probably better. 

LGTM ! 

You can merge your PR using the following workflow:  https://github.com/clouway/teamspace/wiki/Git%20Workflow

**Pull request merged**